### PR TITLE
Fixed Chaplain rolling Anti-Chap Antagonists

### PR DESCRIPTION
**Pull request by ProtozoaPrime**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

## Changelog
:cl:
fix: Chaplains can no longer roll anti-chaplain antagonists
/:cl:

## About The Pull Request
This adds chaplains to the restricted roles for the "choice" antagonists, stopping them from choosing the devil or vampire roles.

## Why It's Good For The Game
Bug fixes are good for the video game

### CODE REVIEW

**Comment by carlopault** [comment]
+1 speedmerge this

File: hippiestation/code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	antag_flag = ROLE_TRAITOR
```
Comment (AnthonyJNgo) [inline hippiestation/code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm]: Are you sure this restrics from devil and other antichaplain antags? It looks like it is just traitor.

File: hippiestation/code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm [hunk 2]
```diff
@@ -3,7 +3,7 @@
...
 	minimum_required_age = 0
 	protected_roles = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Vice Officer")
-	restricted_roles = list("Cyborg")
+	restricted_roles = list("Cyborg", "Chaplain")
```
Comment (AnthonyJNgo) [inline hippiestation/code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm]: I see you added Chaplain here, which is under the Traitor role.

File: hippiestation/code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm [hunk 3]
```diff
@@ -130,16 +130,16 @@
...
 	requirements = list(100,90,80,60,40,30,10,10,10,10)
 	high_population_requirement = 10
-	
+
 /datum/dynamic_ruleset/roundstart/wizard
 	weight = 20
-	
+
 /datum/dynamic_ruleset/roundstart/bloodcult
 	restricted_roles = list("AI", "Cyborg", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Chaplain", "Head of Personnel", "Vice Officer")
```
Comment (AnthonyJNgo) [inline hippiestation/code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm]: Example here, where Chaplain is already listed as a restricted role under bloodcult.

**Comment by AnthonyJNgo** [comment]
Revoked my earlier comments, I'd like @yoyobatty to double check this but it looked fine to me.

**Comment by ProtozoaPrime** [comment]
Oh yeah, this PR is a direct fix for #11966

**Comment by yoyobatty** [comment]
This just stops Chaplain from being able to pick traitor, hivemind, vamp or devil. You need to set it so they can't choose vampire or devil exclusively.

File: hippiestation/code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm (hunk 2 above)
Comment (yoyobatty) [inline hippiestation/code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm]:
```suggestion
	restricted_roles = list("Cyborg")
```

**Review by yoyobatty — CHANGES_REQUESTED** [review]
One thing you could do under line 37-42
```
If(M.mind.assigned_role == "Chaplain")
    to_chat(M, "Error; Cannot select this antag with your current role")
    return
```

**Comment by ProtozoaPrime** [comment]
Gamer moment :sunglasses:

**Review by yoyobatty — APPROVED** [review]
Eh, why not

**Pull request merged**